cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java

```diff
@@ -1032,6 +1037,27 @@ TSStatus processPlanLocally(PhysicalPlan plan) {
       log.setCurrLogTerm(getTerm().get());
       log.setCurrLogIndex(logManager.getLastLogIndex() + 1);
 
+      // Set version for some special plans
+      try {
+        ClusterUtils.setVersionForSpecialPlan(plan, log.getCurrLogIndex());
+      } catch (StorageGroupNotSetException sge) {
+        // if sg not set, sync the meta leader and try again
+        try {
+          ((CMManager) IoTDB.metaManager).syncMetaLeader();
```

Comment:
Holding the lock to send RPCs is a very dangerous design. Can't we put this code after releasing the lock?